src/System.IO.Ports/src/System/IO/Ports/SerialPort.cs

```diff
@@ -0,0 +1,1307 @@
+
+        public bool CtsHolding
+        {
+            get
+            {
+                if (!IsOpen)
+                    throw new InvalidOperationException(SR.Port_not_open);
+                return _internalSerialStream.CtsHolding;
+            }
+        }
+
+        public int DataBits
+        {
+            get
+            { return _dataBits; }
+            set
+            {
+                if (value < MinDataBits || value > MaxDataBits)
+                    throw new ArgumentOutOfRangeException(nameof(DataBits), string.Format(SR.ArgumentOutOfRange_Bounds_Lower_Upper, MinDataBits, MaxDataBits));
+
+                if (IsOpen)
+                    _internalSerialStream.DataBits = value;
+                _dataBits = value;
+            }
+        }
+
+        public bool DiscardNull
+        {
+            get
+            {
+                return _discardNull;
+            }
+            set
+            {
+                if (IsOpen)
+                    _internalSerialStream.DiscardNull = value;
+                _discardNull = value;
+            }
+        }
+
+        public bool DsrHolding
+        {
+            get
+            {
+                if (!IsOpen)
+                    throw new InvalidOperationException(SR.Port_not_open);
+                return _internalSerialStream.DsrHolding;
+            }
+        }
+
+        public bool DtrEnable
+        {
+            get
+            {
+                if (IsOpen)
+                    _dtrEnable = _internalSerialStream.DtrEnable;
+
+                return _dtrEnable;
+            }
+            set
+            {
+                if (IsOpen)
+                    _internalSerialStream.DtrEnable = value;
+                _dtrEnable = value;
+            }
+        }
+
+        // Allows specification of an arbitrary encoding for the reading and writing functions of the port
+        // which deal with chars and strings.  Set by default in the code to System.Text.ASCIIEncoding(), which
+        // is the standard text encoding for modem commands and most of serial communication.
+        public Encoding Encoding
+        {
+            get
+            {
+                return _encoding;
+            }
+            set
+            {
+                if (value == null)
+                    throw new ArgumentNullException(nameof(Encoding));
+
+                // Limit the encodings we support to some known ones.  The code pages < 50000 represent all of the single-byte
+                // and double-byte code pages.  Code page 54936 is GB18030.  Finally we check that the encoding's assembly
+                // is mscorlib, so we don't get any weird user encodings that happen to set a code page less than 50000. 
+                if (!(value is ASCIIEncoding || value is UTF8Encoding || value is UnicodeEncoding || value is UTF32Encoding ||
+                      ((value.CodePage < 50000 || value.CodePage == 54936) && value.GetType().Assembly == typeof(String).Assembly)))
```

Comment:
Is this check correct, given that code pages are in the System.Text.Encoding.CodePages component for .NET Core instead of corelib?